### ISSUE TITLE
Easier way to interface with catch2?

### ISSUE THREAD

**Issue by ziotom78** [tbrown122387/pf#13]
The instructions in the README ask the user to install a few libraries (boost, eigen3, catch2). However, the requirement on catch2 is not explained well:

1. Catch2's homepage explains that the package can be installed by just copying only the header file;
2. People running any Ubuntu-based distribution will likely install it using `sudo apt install catch`;
3. OTOH, the `CMakeLists.txt` provided by this package expects a few `*.cmake` files from the Catch2 distribution, which means that it requires Catch2 to be installed using a non-standard procedure. (No, `apt install` does not install these files where `cmake` can find them`.)

I see a few possible solutions:

1. Include a copy of the Catch2 header file in the source (possible if Catch2's license is compatible with this one: IANAL, so I cannot give advice here). This would be the easiest and cleanest solution;

2. Enable `cmake` to detect the presence of the Catch2 header file when the `*.cmake` files are not found

3. Explain in the README the expected way to install Catch2.

(This issue was raised while carrying on the review for JOSS: https://github.com/openjournals/joss-reviews/issues/2599.)

**Comment by tbrown122387** [comment]
@ziotom78 regarding (3), which `*.cmake` files are you referring to? Are you referring to [this](https://github.com/tbrown122387/pf/blob/master/test/CMakeLists.txt) `CMakeLists.txt` file or [this one?](https://github.com/tbrown122387/pf/blob/master/CMakeLists.txt)

**Comment by ziotom78** [comment]
@tbrown122387 , I was quoting the error message produced by `cmake`:

```
$ cmake ..
CMake Error at CMakeLists.txt:11 (find_package):
  By not providing "FindCatch2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Catch2", but
  CMake did not find one.

  Could not find a package configuration file provided by "Catch2" (requested
  version 2.9.2) with any of the following names:

    Catch2Config.cmake
    catch2-config.cmake

  Add the installation prefix of "Catch2" to CMAKE_PREFIX_PATH or set
  "Catch2_DIR" to a directory containing one of the above files.  If "Catch2"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/tomasi/Documents/work/articles/referee/JOSS/pf/pf/build/CMakeFiles/CMakeOutput.log".
```

Note that I already downloaded and manually installed `catch2/catch.hpp` (following the [instructions on their site](https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#getting-catch2)):
```
$ sudo mkdir -p /usr/include/catch2
$ curl https://raw.githubusercontent.com/catchorg/Catch2/master/single_include/catch2/catch.hpp \
  | sudo tee /usr/include/catch2/catch.hpp > /dev/null
$  head /usr/include/catch2/catch.hpp 
/*
 *  Catch v2.13.1
 *  Generated: 2020-09-07 12:12:38.090364
 *  ----------------------------------------------------------
 *  This file has been merged from multiple headers. Please don't edit it directly
 *  Copyright (c) 2020 Two Blue Cubes Ltd. All rights reserved.
 *
 *  Distributed under the Boost Software License, Version 1.0. (See accompanying
 *  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 */
$
```

**Comment by tbrown122387** [comment]
Catch2 describes two installation options [here](https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#getting-catch2). I prefer to ask users to [install the CMake package](https://github.com/catchorg/Catch2/blob/master/docs/cmake-integration.md#installing-catch2-from-git-repository). I think this will allow everything to build unchanged. @ziotom78

**Comment by ziotom78** [comment]
@tbrown122387 , ok, I believe it's fine, as long as this is explicitly stated in the documentation.

**Comment by tbrown122387** [comment]
closed by e5213149ae4150a8cc384a5f5d84c55e1260664e
